### PR TITLE
Add release workflow and CPI (1.0.1) and CSI (2.3.0-rancher1) charts.

### PR DESCRIPTION
**Pull request by rayandas**
Related: https://github.com/rancher/rancher/issues/35702
After this gets merged the CI will release `rancher-vsphere-cpi-1.0.1.tg` and `rancher-vsphere-csi-2.3.0-rancher1.tgz` and also tag them.
> Note: This is just a mock release. When this gets merged we'll update CPI and CSI to the latest and use the releases inside rancher/charts as well as rancher/rke2-charts

TODO (Do Not Merge before this):
- [ ] Test: Add the tarball urls to rancher/charts and use that to test vSphere CPI and CSI.

### CODE REVIEW

**Comment by PennyScissors** [comment]
@rayandas I don't think these are the right charts, they look like the ones in the 2.5 version line. The charts in here should be what each chart has in the packages/rancher-vsphere-c?i/chartsd directory.
CPI: https://github.com/rancher/charts/tree/dev-v2.6/packages/rancher-vsphere-cpi/charts
CSI: https://github.com/rancher/charts/tree/dev-v2.6/packages/rancher-vsphere-csi/charts

Also, have you tested the ci github workflow in your fork by doing a mock release and installing/validating the released tarball?

**Comment by rayandas** [comment]
I made the changes locally, after test I will update this PR.

**Comment by samkulkarni20** [comment]
@PennyScissors The PR is ready for your review again. Rayan has made the changes you suggested recently & we also tested it with the rancher/charts repo by setting this up as the upstream.

**Review by samkulkarni20 — APPROVED** [review]

**Review by doflamingo721 — DISMISSED** [review]
LGTM

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: vSphere Charts
+
+on:
+  pull_request:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (rayandas) [reply .github/workflows/ci.yaml:6]: Ah! right. sure I'll update this, also rename the workflow.

**Comment by rayandas** [comment]
@PennyScissors Pushed new changes as you suggested.

**Review by samkulkarni20 — APPROVED** [review]

**Pull request merged**